### PR TITLE
Integral double exp

### PR DESCRIPTION
**Pull request by monien**
Working version of simple double exponential with examples with various types of singularities.

### CODE REVIEW

**Comment by monien** [comment]
I would need more details why the compilation is not successful on Ubuntu  Clang to fix the problem.

File: dev/check_examples.sh [hunk 1]
```diff
@@ -185,6 +185,22 @@ then
...
+         exit 1
+     fi
+     echo "$res" | perl -0ne 'if (/result = 0.88622692545275801364908374167057/g) {$found=1; last} END {exit !$found }'
+     if test "$?" != "0";
+     then
+	 echo "FAIL"
+         exit 2
+     fi
+     echo "PASS"
+     exit 0
```
Comment (albinahlback) [inline dev/check_examples.sh:203]: Fix the indentation to 4 spaces. And `elif` needs to be passed with a `then`.

**Comment by albinahlback** [comment]
Looks good! Can you squash the commits before I merge?

**Comment by albinahlback** [comment]
Sorry, I meant squashing the commits you already committed to this branch. In other words, I want you to squash the commits you previously committed before [67f5f76](https://github.com/flintlib/flint/pull/1901/commits/67f5f76067b8dde0c9c25c2f87ddb329adbb89e5) to one single commit. Other commits related to the example should be going in to that commit one as well.

**Comment by albinahlback** [comment]
Just to be clear, I do not want [67f5f76](https://github.com/flintlib/flint/pull/1901/commits/67f5f76067b8dde0c9c25c2f87ddb329adbb89e5) in this PR.

**Comment by albinahlback** [comment]
The git message is not really descriptive. The title of the message should be a short and concise desciption of what have been done (usually no more than 50 characters long), and is usually written in imperative. The body of the message can go into more details of what have been done. Also, perhaps it doesn't make sense to have lines that acts like fixup messages as they are now in one commit (such as "wrong error estimate."), so these lines could just be removed.

**Comment by albinahlback** [comment]
It is useful to check out the history of the code. For instance, descriptive commit messages are very useful when debugging.

**Comment by fredrik-johansson** [comment]
This is a really nice example program to have. Not sure why the checking script is still failing, but will be happy to merge once that is sorted out.

**Comment by albinahlback** [comment]
It doesn't get the expected printout from the example since the return value is 2 from Make.

**Comment by albinahlback** [comment]
(It return 0 if the example went as expected, 3 if the example could not be found, 1 if the example returned with a non-zero value, and 2 if the output is incorrect.)

**Pull request merged**